docs(asset): Add documentation of asset creation on new management API (#361)
eclipse-tractusx/tractusx-edc: docs/samples/Create Asset.md

> How such an environment can be setup locally is documented in [chapter 1](#1-optional---local-setup).

## Table of Content

1. [Optional - Local Setup](#1-optional---local-setup)
2. [Terminology](#2-terminology)
3. [Values](#3-values)
4. [Simple Asset](#4-simple-asset)
5. [Custom Property Asset](#5-custom-property-asset)
6. [Private Property Asset](#6-private-property-asset)
7. [Complex Property Asset](#7-complex-property-asset)

## 1. Optional - Local Setup

## 2. Terminology
| Connector           | New Protocol (DCAT Catalogs) | Description                                                   |
|---------------------|------------------------------|---------------------------------------------------------------|
| Asset Entry         | Asset                        | Contains the Asset ID and its Data Address.                   |
| Contract Definition | Asset Entry / Dataset        | Contains an Asset () that is offered and covered by a Policy. |

## 3. Values
| Key      | Description |
|----------|-------------|
| @context |             |
| @vocab   |             |
| edc      |             |

## 4. Simple Asset

````json
{

>   "https://w3id.org/edc/v0.0.1/ns/asset":{

Reviewer: I know, it's still a Draft, but I have a suggestion already: Add spaces between the colons and the curved brackets in all the JSON examples:

```suggestion
  "https://w3id.org/edc/v0.0.1/ns/asset": {
```

Author: Well, I specifically made it a draft to get early feedback, so no worries.
Applied the suggestion.

>   "https://w3id.org/edc/v0.0.1/ns/asset":{

Reviewer 2: generally, you don't have to specify the namespace for every property, and the context could be defined once at the root:
```json
{
  "@context":{
    "@vocab":"https://w3id.org/edc/v0.0.1/ns/",
    "edc":"https://w3id.org/edc/v0.0.1/ns/"
  },
  "asset": {
    "@type": "DataAddress",
    "@id": "some-asset-id",
    "properties": {
        "foo": "bar"
    }
  },
  "dataAddress": {
    "@type": "DataAddress",
    "type": "test-type",
    "keyName": "test-key-name"
  }
}
```

Author: Moved the context to the root in all examples.

    "@context":{
      "@vocab":"https://w3id.org/edc/v0.0.1/ns/",
      "edc":"https://w3id.org/edc/v0.0.1/ns/"
    },
    "@type":"https://w3id.org/edc/v0.0.1/ns/Asset",
    "@id":"some-asset-id"
  },
  "https://w3id.org/edc/v0.0.1/ns/dataAddress":{
    "@context":{
      "@vocab":"https://w3id.org/edc/v0.0.1/ns/",
      "edc":"https://w3id.org/edc/v0.0.1/ns/"
    },
    "@type":"https://w3id.org/edc/v0.0.1/ns/DataAddress",
    "https://w3id.org/edc/v0.0.1/ns/https://w3id.org/edc/v0.0.1/ns/type":"test-type",

>     "https://w3id.org/edc/v0.0.1/ns/https://w3id.org/edc/v0.0.1/ns/keyName":"test-key-name"

Reviewer 2: duplicate namespace

| Connector           | New Protocol (DCAT Catalogs) | Description                                                   |
|---------------------|------------------------------|---------------------------------------------------------------|
| Asset Entry         | Asset                        | Contains the Asset ID and its Data Address.                   |
| Contract Definition | Asset Entry / Dataset        | Contains an Asset () that is offered and covered by a Policy. |

Generally, this documentation uses the connector terminology unless otherwise specified.

## 3. Values

Since some keys which are required in requests for the new management API aren't self-explanatory when you first see them, a short explanation is given below.

| Key      | Description |
|----------|-------------|
| @context |             |
| @vocab   |             |
| edc      |             |

## 4. Simple Asset

The first use-case is to generate a very simple asset, that only contains the minimum in terms of information.
For this we need both an asset and a data address, which together form an asset entry.

> ````json

Author: Changed it in the whole file.

# Create Asset

This document will showcase how to create an asset with the new management API.

> Please note: Before running the examples the corresponding environment variables must be set.
> How such an environment can be setup locally is documented in [chapter 1](#1-optional---local-setup).

## Table of Content

1. [Optional - Local Setup](#1-optional---local-setup)
2. [Terminology](#2-terminology)
3. [Values](#3-values)
4. [Simple Asset](#4-simple-asset)
5. [Custom Property Asset](#5-custom-property-asset)
6. [Private Property Asset](#6-private-property-asset)
7. [Complex Property Asset](#7-complex-property-asset)

## 1. Optional - Local Setup

> TODO link to a 0.4.x example setup.

Reviewer: When will this be available?

## 2. Terminology

> With the migration to a new protocol and with it new specifications, DCAT Catalogs in the context of assets, there is a need to clarify the terminology that is used.

Reviewer: Additionally, can we provide links to the new protocol and the specification?

```suggestion
With the migration to the new protocol and with its new specification, DCAT Catalogs in the context of assets, there is a need to clarify the terminology that is used.
```

|---------------------|------------------------------|------------------------------------------------------------|
| Asset Entry         | Asset                        | Contains the Asset ID and its Data Address.                |
| Contract Definition | Asset Entry / Dataset        | Contains an Asset that is offered and covered by a Policy. |

Generally, this documentation uses the connector terminology unless otherwise specified.

## 3. Values

Since some keys which are required in requests for the new management API aren't self-explanatory when you first see them, a short explanation is given below.
These keys are generally part of the JSON-LD spec and serve as identification on a larger scope.

| Key      | Description                                                                                                                          |
|----------|--------------------------------------------------------------------------------------------------------------------------------------|
| @context | JSON-LD context aims to enable the local usage of compact and meaningful names which are later still identifiable in a larger scope. |

> | @vocab   | The value of `@vocab` is added as prefix to any key consisting of a plain string without any other prefix.                           |

Reviewer: Whats the purpose of this?

> | edc      | `edc` defines a specific prefix for the connector.                                                                                   |

Reviewer: Whats the purpose of this prefix?

> | @id      | `@id` contains a distinct identifier enabling global references to this field, also called a Node.                                   |

Reviewer: field or object?

> | Key      | Description                                                                                                                          |
> |----------|--------------------------------------------------------------------------------------------------------------------------------------|
> | @context | JSON-LD context aims to enable the local usage of compact and meaningful names which are later still identifiable in a larger scope. |
> | @vocab   | The value of `@vocab` is added as prefix to any key consisting of a plain string without any other prefix.                           |
> | edc      | `edc` defines a specific prefix for the connector.                                                                                   |
> | @id      | `@id` contains a distinct identifier enabling global references to this field, also called a Node.                                   |
> | @type    | Specifies the path to the schema which describes this Node.                                                                          |

Reviewer: ```suggestion
| Key        | Description                                                                                                                          |
|------------|--------------------------------------------------------------------------------------------------------------------------------------|
| `@context` | JSON-LD context aims to enable the local usage of compact and meaningful names which are later still identifiable in a larger scope. |
| `@vocab`   | The value of `@vocab` is added as prefix to any key consisting of a plain string without any other prefix.                           |
| `edc`      | `edc` defines a specific prefix for the connector.                                                                                   |
| `@id`      | `@id` contains a distinct identifier enabling global references to this field, also called a Node.                                   |
| `@type`    | Specifies the path to the schema which describes this Node.                                                                          |
```

> Please note: The namespaces in `@vocab` and `edc` currently are only placeholders and do not lead to any JSON-LD context definition.
> This will change at a later date.

## 4. Simple Asset

The first use-case is to generate a very simple asset, that only contains the minimum in terms of information.
For this we need both an asset and a data address, which together form an asset entry.

```json
{
  "@context": {
    "@vocab": "https://w3id.org/edc/v0.0.1/ns/",
    "edc": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "edc:asset": {

>     "@type": "edc:asset",

Reviewer 2: ```suggestion
    "@type": "Asset",
```

    "@id": "some-asset-id"
  },
  "edc:dataAddress": {
    "@type": "edc:dataAddress",
    "type": "test-type",

>     "keyName": "test-key-name"

Reviewer: In my opinion  using a real type of dataAddress would be more beneficial. E.g. HttpData

  }
}
```

The asset entry is then sent to the management API of the connector through a POST request.

> **CON_DATAMGMT_URL** specifies the URL of the management API.

Reviewer: ```suggestion
**CON_DATAMGMT_URL** specifies the base URL of the connector context, where the management API is available.
```

                  "edc:version": "0.2.1",
                  "contenttype": "application/json"
                }
              },
              "edc:dataAddress": {
                "@type": "edc:dataAddress",
                "type": "test-type",
                "keyName": "test-key-name"
              }
            }' \
    -s -o /dev/null -w 'Response Code: %{http_code}\n'
```

## 6. Private Property Asset

A new addition are the private properties.
Private properties will not be sent through the dataplane and are only accessible via the management API.
This enables the storage of additional information pertaining the asset, that is not relevant for the consumer, but is nonetheless useful for the provider.
Private properties are stores inside the `privateProperties` field.

```json
{
  "@context": {
    "@vocab": "https://w3id.org/edc/v0.0.1/ns/",
    "edc": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "edc:asset": {
    "@type": "edc:asset",
    "@id": "some-asset-id",
    "properties": {
      "name": "some-asset-name",
      "description": "some description",
      "edc:version": "0.2.1",
      "contenttype": "application/json"
    },

>     "edc:privateProperties": {

Reviewer: why do we have here the `edc` prefix and on the "normale" properties not?

Reviewer: Is it, because the edc stores everything which is prefixed with `edc:` separately?

    --header 'X-Api-Key: password' \
    --header 'Content-Type: application/json' \
    --data '{
              "@context": {
                "@vocab": "https://w3id.org/edc/v0.0.1/ns/",
                "edc": "https://w3id.org/edc/v0.0.1/ns/"
              },
              "edc:asset": {
                "@type": "edc:asset",
                "@id": "some-asset-id",
                "properties": {
                  "name": "some-asset-name",
                  "description": "some description",
                  "edc:version": "0.2.1",
                  "contenttype": "application/json"
                },
                "edc:privateProperties": {
                  "test-prop": "test-val"
                }
              },
              "edc:dataAddress": {
                "@type": "edc:dataAddress",
                "type": "test-type",
                "keyName": "test-key-name"
              }
            }' \
    -s -o /dev/null -w 'Response Code: %{http_code}\n'
```

## 7. Complex Property Asset

Besides primitive datatypes, complex JSON objects can also be stored inside the properties.
An example is the `payload` field and its contents.

> Here the `payload` contains the name and age of a person.

Reviewer: Can it be anything? Because then I wouldn't use the term `payload`. This sounds like, that I can store the to be transmitted data within this field.
